src/main/ruby/core/argf.rb

```diff
@@ -59,6 +59,7 @@ class ARGFClass
     def initialize(argv = ARGV, *others)
       @argv = argv.equal?(ARGV) ? ARGV : [argv, *others]
       @lineno = 0
+      @last_lineno = 0
```

Comment:
Does `@last_lineno` need to be a separate variable?
It seems to always have the same value as `@lineno`.

Reply:
Except that setting `$.` sets `ARGF.last_lineno` but not `ARGF.lineno`. I had to read the MRI source quite carefully to figure out why there was such an odd relationship between `$.` and `ARGF.lineno` so that I could actually match the behaviour.

Reply:
Oh wow, that's weird indeed. Could you add a comment about it above the definition of `@last_lineno`?

Reply:
Will do.